### PR TITLE
Bump testng from 6.10 to 6.14.3

### PR DESCRIPTION
**Pull request by kokosing**
Bump testng from 6.10 to 6.14.3

### CODE REVIEW

**Comment by kokosing** [comment]
Related to https://github.com/prestosql/presto/pull/1636, 7.0.0 looks unstable.

File: presto-tests/src/main/java/io/prestosql/tests/AbstractTestAggregations.java [hunk 1]
```diff
@@ -847,8 +847,8 @@ public void testGroupByNanArray()
         List<MaterializedRow> actualRows = actual.getMaterializedRows();
         assertEquals(actualRows.size(), 1);
         assertTrue(Double.isNaN(((List<Double>) actualRows.get(0).getField(0)).get(0)));
-        assertEquals(((List<Double>) actualRows.get(0).getField(0)).get(1), 2.0);
-        assertEquals(((List<Double>) actualRows.get(0).getField(0)).get(2), 3.0);
+        assertEquals((double) ((List<Double>) actualRows.get(0).getField(0)).get(1), 2.0);
+        assertEquals((double) ((List<Double>) actualRows.get(0).getField(0)).get(2), 3.0);
```
Comment (findepi) [inline presto-tests/src/main/java/io/prestosql/tests/AbstractTestAggregations.java]: Why is this needed?
(do we needed this elsewhere?)

File: pom.xml [hunk 2]
```diff
@@ -53,7 +53,7 @@
         <dep.drift.version>1.14</dep.drift.version>
         <dep.joda.version>2.10.3</dep.joda.version>
         <dep.tempto.version>166</dep.tempto.version>
-        <dep.testng.version>6.10</dep.testng.version>
+        <dep.testng.version>6.14.3</dep.testng.version>
```
Comment (findepi) [inline pom.xml]: What good does it bring?

File: presto-tests/src/main/java/io/prestosql/tests/AbstractTestAggregations.java (hunk 1 above)
Comment (kokosing) [reply presto-tests/src/main/java/io/prestosql/tests/AbstractTestAggregations.java]: It looks like that they changed API and otherwise it fails the compilation. I compiled all project and no other changes were needed.

File: pom.xml (hunk 2 above)
Comment (kokosing) [reply pom.xml]: Not sure, testng release notes are pretty vague https://github.com/cbeust/testng/blob/master/CHANGES.txt. 

There are couple bug fixes, and maybe new bugs as well.
